src/app/qgsjoindialog.cpp

```diff
@@ -42,6 +42,8 @@ QgsJoinDialog::QgsJoinDialog( QgsVectorLayer *layer, QList<QgsMapLayer *> alread
 
   mTargetFieldComboBox->setLayer( mLayer );
 
+  mDynamicFormCheckBox->setToolTip( tr( "This option allows values of the joined fields to be automatically updated when the \"Target Field\" is updated" ) );
```

Comment:
Maybe replace the first "updatet" with reloaded or adjusted (better suggestions welcome) and the second one with "changed" to avoid confusion with the concept of an update in db speak where it's changed in the data provider? cc @DelazJ